x-pack/plugins/ml/public/ui_actions/apply_entity_filters_action.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { createAction } from '../../../../../src/plugins/ui_actions/public';
+import { MlCoreSetup } from '../plugin';
+import { Filter, FilterStateStore } from '../../../../../src/plugins/data/common';
+import { ANOMALY_EXPLORER_EMBEDDABLE_TYPE, ExplorerFieldSelectionContext } from '../embeddables';
+import { CONTROLLED_BY_EXPLORER_FILTER } from './constants';
+
+export const APPLY_ENTITY_FIELD_FILTERS_ACTION = 'applyEntityFieldFiltersAction';
+
+export function createApplyEntityFieldFiltersAction(
+  getStartServices: MlCoreSetup['getStartServices']
+) {
+  return createAction<ExplorerFieldSelectionContext>({
+    id: 'apply-entity-field-filters',
+    type: APPLY_ENTITY_FIELD_FILTERS_ACTION,
+    getIconType(context: ExplorerFieldSelectionContext): string {
+      return 'filter';
+    },
+    getDisplayName() {
+      return i18n.translate('xpack.ml.actions.applyEntityFieldsFiltersTitle', {
+        defaultMessage: 'Filter for value',
+      });
+    },
+    async execute({ data }) {
+      if (!data) {
+        throw new Error('No entities provided');
+      }
+      const [, pluginStart] = await getStartServices();
+      const filterManager = pluginStart.data.query.filterManager;
+
+      filterManager.addFilters(
+        data
+          .filter((d) => d.operation === '+')
```

Comment:
you can you a constant for the operator check

Reply:
Updated here https://github.com/elastic/kibana/pull/94396/commits/50ce97acc22dfb442a067d88f40c9668d671cc21